content/specs/handle.md

```diff
@@ -0,0 +1,119 @@
+
+A reference regular expression (regex) for the handle syntax is:
+
+```
+/^([a-zA-Z0-9]([a-zA-Z0-9-]{0,61}[a-zA-Z0-9])?\.)+[a-zA-Z]([a-zA-Z0-9-]{0,61}[a-zA-Z0-9])?$/
+```
+
+### Additional Non-Syntax Restrictions
+
+“Reserved” top-level domains should not fail syntax validation (eg, in atproto Lexicon validation), but they must immediately fail any attempt at registration, resolution, etc. See also: [https://en.wikipedia.org/wiki/Top-level_domain#Reserved_domains](https://en.wikipedia.org/wiki/Top-level_domain#Reserved_domains)
+
+`.local` hostnames (for mDNS on local networks) should not be used in atproto.
+
+The `.onion` TLD is a special case for Tor protocol hidden services. Resolution of handles via Tor would require ecosystem-wide support, so they are currently disallowed.
+
+To summarize the above, the initial list of disallowed TLDs includes:
+
+- `.local`
+- `.arpa`
+- `.invalid`
+- `.localhost`
+- `.internal`
+- `.onion`
+
+The `.test` TLD is intended for examples, testing, and development. It may be used in atproto development, but should fail in real-world environments.
+
+### Examples
+
+Syntactically valid handles (which may or may not have existing TLDs): 
+
+```
+jay.bsky.social
+8.cn
+name.t--t        // not a real TLD, but syntax ok
+XX.LCS.MIT.EDU
+a.co
+xn--notarealidn.com
+xn--fiqa61au8b7zsevnm8ak20mc4a87e.xn--fiqs8s
+xn--ls8h.test
+example.t        // not a real TLD, but syntax ok
+```
+
+Invalid syntax:
+
+```
+jo@hn.test
+💩.test
+john..test
+xn--bcher-.tld
+john.0
+cn.8
+www.masełkowski.pl.com
+org
+name.org.
+```
+
+Valid syntax, but must always fail resolution due to other restrictions:
+
+```
+2gzyxa5ihm7nsggfxnu52rck2vv4rvmdlkiu3zzui5du4xyclen53wid.onion
+laptop.local
+blah.arpa
+```
+
+### Usage and Implementation Guidelines
+
+Handles may be prefixed with the “at” symbol (like `@jay.bsky.team`) in user interfaces, but this is not a valid syntax for a handle in records, APIs, and other back-end contexts.
+
+Internationalized Domain Names (”IDN”, or “punycode”) are not directly relevant to the low-level handle syntax. In their encoded form, IDNs are already valid hostnames, and thus valid handles. Such handles must be stored and transmitted in encoded ASCII form. Handles that “look like” IDNs, but do not parse as valid IDNs, are valid handles, just as they are valid hostnames. Applications may, optionally, parse and display IDN handles as Unicode.
+
+Handles are not case-sensitive, which means they can be safely normalized from user input to lower-case (ASCII) form. Only normalized (lower-case) handles should be stored in records or used in outbound API calls. Applications should not preserve user-provided case information and attempt to display handles in anything other than lower-case. For example, the handle input string `[BlueskyWeb.xyz](http://www.BlueskyWeb.xyz)`  should be normalized, stored, and displayed as `blueskyweb.xyz`.
```

Comment:
Should this be a markdown link or just `BlueskyWeb.xyz` ?

Reply:
Also applies to links further down (paypal, jay.bsky.social)

Reply:
thanks!